src/commands/cmd_server.cc

```diff
@@ -1059,8 +1060,9 @@ class CommandRestore : public Commander {
       ttl_ms_ -= now;
     }
 
-    RDB rdb(svr->storage, conn->GetNamespace(), args_[3]);
-    auto s = rdb.Restore(args_[1], ttl_ms_);
+    auto stream_ptr = std::make_shared<RdbStringStream>(args_[3]);
```

Comment:
Could you make a little explanation here about where this pointer need to be "share" for undetermined lifetime? Otherwise we'd better to use `unique_ptr` : )

Reply:
Yes, it's better to use unique_ptr.